src/utils/github.ts

```diff
@@ -0,0 +1,127 @@
+import { Octokit } from 'octokit';
+import axios from 'axios';
+import pLimit from 'p-limit';
+
+import { log } from '../log.js';
+
+import path from 'path';
+import fs from 'fs';
+
+const MULTITHREADING_LIMIT = 10;
```

Comment:
nit: can we rename this to `CONCURRENT_OPERATIONS_LIMIT`? The current name is clear, but it's not `threads` that we are referring here